boot/bootutil/src/loader.c

```diff
@@ -1221,21 +1221,71 @@ boot_swap_if_needed(int *out_swap_type)
 }
 
 /**
+ * Finalize some administration work, after swapping was done:
+ *  - set copy done flag on demand
+ *  - decide wether image headers must be reloaded or not
+ *
+ * @param swap_type         The type of the previously done swapping operation
+ * @param reload_headers    Flag to indicates do we need to reload the image's
+ *                          headers
+ */
+static void
+boot_finalize_swap(int swap_type, bool *reload_headers)
```

Comment:
This can't be correct without also knowing the image ID number, unless I am missing something.